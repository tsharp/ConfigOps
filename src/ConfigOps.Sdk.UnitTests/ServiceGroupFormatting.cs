using ConfigOps.Sdk.Api.v1alpha1;
using ConfigOps.Sdk.Serialization;
using Snapshooter.Xunit;

namespace ConfigOps.Sdk.UnitTests
{
    public class ServiceGroupFormatting
    {
        [Fact]
        public void Validate_AwsServiceGroup_YamlFormat()
        {
            ServiceGroup serviceGroup = new ServiceGroup()
            {
                Metadata =
                {
                    Name = "service-example-uswest1",
                    Namespace = "amazon-aws"

                },
                Data =
                {
                    CloudProvider = "amazon-aws/cloud-config",
                    CloudProviderRegion = "amazon-aws/us-west-1",
                    ExtensionData =
                    {
                        { "secretStoreXyz", "abcd" }
                    }
                }
            };

            serviceGroup
                .ObjectToYaml()
                .MatchSnapshot();
        }
    }
}
