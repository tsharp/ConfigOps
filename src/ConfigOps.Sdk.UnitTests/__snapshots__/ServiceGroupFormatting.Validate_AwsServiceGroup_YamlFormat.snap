data:
  secretStoreXyz: abcd
apiVersion: config.kuipersys.com/v1alpha1
kind: ServiceGroup
metadata:
  name: service-example-uswest1
  namespace: amazon-aws
  inheritedConfigs:
